transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/DefaultScheduler.java

```diff
@@ -60,24 +61,33 @@
    * @param event The event itself which needs to be logged with additional information.
    */
   @Override
-  public void schedule(TransportContext transportContext, EventInternal event) {
+  public void schedule(
+      TransportContext transportContext, EventInternal event, TransportScheduleCallback callback) {
     executor.execute(
         () -> {
-          TransportBackend transportBackend =
-              backendRegistry.get(transportContext.getBackendName());
-          if (transportBackend == null) {
-            LOGGER.warning(
-                String.format(
-                    "Transport backend '%s' is not registered", transportContext.getBackendName()));
-            return;
+          try {
+            TransportBackend transportBackend =
+                backendRegistry.get(transportContext.getBackendName());
+            if (transportBackend == null) {
+              String errorMsg =
+                  String.format(
+                      "Transport backend '%s' is not registered",
+                      transportContext.getBackendName());
+              LOGGER.warning(errorMsg);
+              callback.onSchedule(new IllegalArgumentException(errorMsg));
+              return;
+            }
+            EventInternal decoratedEvent = transportBackend.decorate(event);
+            guard.runCriticalSection(
+                () -> {
+                  eventStore.persist(transportContext, decoratedEvent);
+                  workScheduler.schedule(transportContext, 1);
+                  return null;
+                });
+            callback.onSchedule(null);
+          } catch (Exception e) {
+            callback.onSchedule(e);
```

Comment:
Before this change the exception would have been logged, can you log it here to preserve this behavior?

Reply:
Done.